data_structures/linked_list/linked_list.dart

```diff
@@ -0,0 +1,40 @@
+class Node {
+  int value;
+  Node next = null;
+  Node(this.value);
+}
+
+class LinkedList {
+  Node head;
+  Node tail;
```

Comment:
[head|tail| is most often referencing to [<first item>|<rest of the list>]. 
In this case it's accumulator for reference to last element of the list.
It may be confusing - you should rename it and probably make private.

Reply:
My sense is that `head` should always be the very first element in the collection and that `tail` should always be the very last element in the collection.  [head, x, y, z, tail]

Reply:
I'm basing mostly on my [haskell experience here](http://zvon.org/other/haskell/Outputprelude/tail_f.html).
In functional languages it's popular to use `tail` like that - https://en.wikipedia.org/wiki/CAR_and_CDR
Anyway code is right now little bit more clear to me.